psydac/core/bsplines.py

```diff
@@ -333,7 +398,8 @@ def collocation_matrix(knots, degree, periodic, normalization, xgrid):
     # Fill in non-zero matrix values
     for i,x in enumerate( xgrid ):
         span  =  find_span( knots, degree, x )
-        basis = basis_funs( knots, degree, x, span )
+        basis = np.zeros(degree + 1)
```

Comment:
Shouldn't we move this array allocation outside the for loop?

Reply:
Yes, I think auto piloted and allocated it directly. This is fixed in c9eac22